### ISSUE TITLE
Make fancier MIN_VERSION_base fallback

### ISSUE THREAD

**Issue by treeowl** [haskell/containers#131]
We currently use the extremely pessimistic

```haskell
#define MIN_VERSION_base(major1,major2,minor) 0
```

Now that the definition is hidden away in an include file and not copied everywhere, we can make it much less so. The below is an optimistic version, but we could use a mildly pessimistic version instead to be really safe.
```haskell
#ifndef MIN_VERSION_base
-- These are not precise, but allow the module to be built without cabal.
#if __GLASGOW_HASKELL__ >= 709
#define MIN_VERSION_base(major1,major2,minor) (((major1) <= 4) && ((major2) <= 8))
#elif __GLASGOW_HASKELL__ >= 707
#define MIN_VERSION_base(major1,major2,minor) (((major1) <= 4) && ((major2) <= 7))
#elif __GLASGOW_HASKELL__ >= 705
#define MIN_VERSION_base(major1,major2,minor) (((major1) <= 4) && ((major2) <= 6))
#else
#define MIN_VERSION_base(major1,major2,minor) (0)
#endif
#endif
```

**Comment by foxik** [comment]
Good idea!

We should probably make the conditions look like
```Haskell
(((major1) < 4) || ((major1) == 4 && (major2) <= 8))
```
even if I doubt ``major1==3`` case will be useful :-)

Could you create a pull request please?

**Comment by treeowl** [comment]
Yes, you're absolutely right. I had the nagging feeling that I was being sloppy there, and indeed I was. I should be able to put together a pull request within the next day.

**Comment by treeowl** [comment]
Fixed by #133
